Lib/test/test_tarfile.py

```diff
@@ -3709,6 +3785,75 @@ def test_parent_symlink(self):
             with self.check_context(arc.open(), 'data'):
                 self.expect_file('parent/evil')
 
+    @symlink_test
+    @os_helper.skip_unless_symlink
+    def test_realpath_limit_attack(self):
+        # (CVE-2025-4517)
+
+        with ArchiveMaker() as arc:
+            # populate the symlinks and dirs that expand in os.path.realpath()
+            # The number is chosen so that in common cases, the unexpanded
+            # path fits in PATH_MAX, but it overflows when the final symlink
+            # is expanded
+            if sys.platform == 'darwin':
+                component = 'd' * 55
+            elif sys.platform == 'win32':
+                component = 'd' * 25
+            else:
+                component = 'd' * 247
+            steps = "abcdefghijklmnop"
+            path = ""
+            step_path = ""
+            for i in steps:
+                arc.add(os.path.join(path, component), type=tarfile.DIRTYPE,
+                        mode='drwxrwxrwx')
+                arc.add(os.path.join(path, i), symlink_to=component)
+                path = os.path.join(path, component)
+                step_path = os.path.join(step_path, i)
+            # create the final symlink that exceeds PATH_MAX and simply points
+            # to the top dir.
+            # this link will never be expanded by
+            # os.path.realpath(strict=False), nor anything after it.
+            linkpath = os.path.join(*steps, "l"*254)
+            parent_segments = [".."] * len(steps)
+            arc.add(linkpath, symlink_to=os.path.join(*parent_segments))
+            # make a symlink outside to keep the tar command happy
+            arc.add("escape", symlink_to=os.path.join(linkpath, ".."))
+            # use the symlinks above, that are not checked, to create a hardlink
+            # to a file outside of the destination path
+            arc.add("flaglink", hardlink_to=os.path.join("escape", "flag"))
+            # now that we have the hardlink we can overwrite the file
+            arc.add("flaglink", content='overwrite')
+            # we can also create new files as well!
+            arc.add("escape/newfile", content='new')
+
+        with self.check_context(arc.open(), filter='fully_trusted',
+                                check_flag=False):
+            if sys.platform == 'win32':
+                self.expect_exception(FileNotFoundError)
```

Comment:
Let's say it's weird & move on?
```suggestion
                self.expect_exception((FileNotFoundError, FileExistsError))
```
This *is* supposed to be a situation where a file exists but doesn't, depending on how you look at it. A `FileExistsError` from `os.makedirs(upperdirs, exist_ok=True)` sort of makes sense; fixing it to make more sense is not really in scope here.